forgot_password/handlers.py

```diff
@@ -102,98 +143,269 @@ def forgot_password(email):
-                return skygear.Response(body, content_type='text/html')
-            user_record = userutil.get_user_record(c, user.id)
 
-        template_params = {
+def send_notification_email(template_provider,
+                            user,
+                            user_record,
+                            settings,
+                            smtp_settings,
+                            notification_email_settings):
+    if not smtp_settings.host:
+        logger.error('Mail server is not configured. '
+                     'Ignore sending notification email')
+    else:
+        url_prefix = settings.url_prefix
+        if url_prefix.endswith('/'):
+            url_prefix = url_prefix[:-1]
+
+        email_params = {
+            'appname': settings.app_name,
+            'url_prefix': url_prefix,
+            'email': user.email,
+            'user_id': user.id,
             'user': user,
             'user_record': user_record,
-            'code': code,
-            'user_id': user_id,
         }
 
-        if request.method == 'POST':
-            password = request.values.get('password')
-            if not password:
-                return reset_password_response(
-                    error='Password cannot be empty.',
-                    **template_params
-                )
+        text = template_provider.\
+            get_template('notification_email_text').\
+            render(**email_params)
+        html = template_provider.\
+            get_template('notification_email_html').\
+            render(**email_params)
+
+        sender = notification_email_settings.sender
+        reply_to = notification_email_settings.reply_to
+        subject = notification_email_settings.subject
+
+        try:
+            mailer = email_util.Mailer(
+                smtp_host=smtp_settings.host,
+                smtp_port=smtp_settings.port,
+                smtp_mode=smtp_settings.mode,
+                smtp_login=smtp_settings.login,
+                smtp_password=smtp_settings.password,
+            )
+            mailer.send_mail(sender, user.email, subject, text,
+                             html=html, reply_to=reply_to)
+            logger.info('Successfully sent notification email '
+                        'to user.')
+        except Exception as ex:
+            logger.error('An error occurred sending notification '
+                         'email to user: {}'.format(str(ex)))
 
-            if password != request.values.get('confirm'):
-                return reset_password_response(
-                    error='Confirm password does not match new password.',
-                    **template_params
-                )
 
-            with conn() as c:
-                userutil.set_new_password(c, user.id, password)
-                logger.info('Successfully reset password for user.')
+def validate_reset_password_request_parameters(db_connection, request):
+    """
+    Validation of reset password request parameters
```

Comment:
I think a better style for function docstring is to use an imperative form with a period at the end. (i.e. “Validates reset password request parameters.”)